src/components/drag-and-drop/draggable-context/draggable-context.js

```diff
@@ -0,0 +1,142 @@
+import React from 'react';
+import { DragDropContext } from 'react-dnd';
+import PropTypes from 'prop-types';
+import HTML5Backend from 'react-dnd-html5-backend';
+import ItemTargetHelper from './../../../utils/helpers/dnd/item-target';
+
+/**
+ * A draggable context component
+ *
+ * == How to use a draggable context in a component:
+ *
+ * In your file
+ *
+ *   import { DraggableContext, WithDrop, WithDrag } from 'carbon/lib/components/drag-and-drop'
+ *
+ * A draggable context is used to define an area in the page where drag and drop can be used on
+ * one or more elements (you also need to use WithDrop and WithDrag):
+ *
+ *   <DraggableContext onDrag={ onItemMoved }>
+ *     <ol>
+ *       {
+ *         items.map((item, index) => {
+ *           return (
+ *             <WithDrop index={ index }>
+ *               <li>
+ *                 <WithDrag><span>{ item.content }</span></WithDrag>
+ *               </li>
+ *             </WithDrop>
+ *           );
+ *         })
+ *       }
+ *     </ol>
+ *   </DraggableContext>
+ *
+ * @class DraggableContext
+ * @constructor
+ */
+class DraggableContext extends React.Component {
+
+  static propTypes = {
+    /**
+     * Callback function for when an item has been dragged
+     * e.g. to update data in a store
+     */
+    onDrag: PropTypes.func.isRequired
+  }
+
+
+  /**
+   * Defines a context object for child components of the draggable context component.
+   * https://facebook.github.io/react/docs/context.html
+   *
+   * @property childContextTypes
+   * @type {Object}
+   */
+  static childContextTypes = {
+    dragAndDropOnDrag: PropTypes.func, // Callback for when order is changed
+    dragAndDropBeginDrag: PropTypes.func, // Callback for when dragging begins
+    dragAndDropEndDrag: PropTypes.func, // Callback for when dragging ends
+    dragAndDropHover: PropTypes.func, // Callback for when a hover is triggered
+    dragAndDropActiveIndex: PropTypes.number // Tracks the currently dragged index
+  }
+
+  /**
+   * Returns this draggable context properties to child components.
+   *
+   * @method getChildContext
+   * @return {void}
+   */
+  getChildContext() {
+    return {
+      dragAndDropOnDrag: this.handleDrag,
+      dragAndDropBeginDrag: this.handleBeginDrag,
+      dragAndDropEndDrag: this.handleEndDrag,
+      dragAndDropHover: this.handleHover,
+      dragAndDropActiveIndex: this.state.activeIndex
+    };
+  }
+
+  state = {
+    activeIndex: null // {Number} tracks the currently dragged index
+  }
+
+  /**
+   * A callback for when hover is triggered
+   *
+   * @method Hover
```

Comment:
I think our intention with comments was not to use the `@method` parameter as it's obvious from the method name (throughout code)